### ISSUE TITLE
Recordedit using incorrect validation for float4 column type

### ISSUE THREAD

**Issue by karlcz** [informatics-isi-edu/chaise#1872]
I have float4 columns and recordedit is complaining about values like `123.5`, showing a red box around the input field and tooltip like `Please select a valid value. The two nearest valid values are 123 and 124.`

**Comment by jrchudy** [comment]
I can only reproduce this "validation" state when typing a float into an integer input field. I looked through each HTML document to verify that the integer validator is only on integer (`int2`, `int4`, and `int8`) inputs. Anything marked as `float4`, `float8`, and `numeric` are treated as decimal inputs with a float validator.

Could you point to the recordedit page that you were having this issue on?

**Comment by karlcz** [comment]
On synapse-dev, go to anything in Zebrafish:Image%20Region table where
the status column says "processed" so there are fractional values in a
bunch of columns near the bottom with "Min" or "Max" in their column
names.

When I open up recordedit on any of these, I see many red-outlined
fields.

**Comment by jrchudy** [comment]
I can reproduce this bug on FF on MacOS. It looks like FF is showing an HTML validation error by highlighting the field and popping up the "different style tooltip" with the validation error message from above (`Please select a valid value. The two nearest valid values are 123 and 124.`)

I can actually see this same validation error on chrome, but there is no "highlight" around the box that notifies the user there is a validation error (according to HTML not our app).

A simple solution for this incorrect validation error reporting, is to add `step` to the HTML tag. Step works like the following:
```
step="any" will allow any decimal.
step="1" will allow no decimal.
step="0.5" will allow 0.5; 1; 1.5; ...
step="0.1" will allow 0.1; 0.2; 0.3; 0.4; ...
```


This did turn up another issue that I hadn't noticed before which is that chrome does not show any UI change for invalid float input values. Furthermore, regardless of the browser, the float validation I wrote in `angular` is not working at all. If you type `123.5.6` into a float input, no validation error message is shown and the page can't be submitted. FF does highlight the input for you though (which I mentioned above).

**Comment by jrchudy** [comment]
The HTML validation part of this issue has been fixed. Float inputs have `step="any"` attached to them. Fixed in [this commit](https://github.com/informatics-isi-edu/chaise/commit/d5b5ec3613aeef9ec90c4addbf389761d3e7b49b).

**Comment by jrchudy** [comment]
Further changes have been merged into master to improve the validation functionality of float inputs and integer inputs. We were relying on a combination of HTML validators and angularJS validators. This caused some bugs when the HTML validators invalidated the values which were then not passed to the angular validators to trigger the error message conditionals.

Changes were made to both float and integer input validators to remove the HTML validators and rely solely on angularJS ones. Changes were merged in PR #1874 .

Float and integer inputs will now properly show an error when the user types a value including alphabetic characters as well as a numeric string that doesn't represent either numeric type (`1.1.1` for example).
